src/app/routes/app-routes.tsx

```diff
@@ -201,3 +198,20 @@ export function AppRoutes() {
     </Routes>
   );
 }
+
+export function AppRoutes() {
+  const hasStateRehydrated = useHasStateRehydrated();
+  const hasResponded = useHasUserRespondedToAnalyticsConsent();
+
+  if (!hasStateRehydrated) return <LoadingSpinner />;
+
+  if (!hasResponded) {
+    return (
+      <Routes>
+        <Route path={RouteUrls.RequestDiagnostics} element={<AllowDiagnosticsPage />} />
+        <Route path="*" element={<Navigate replace to={RouteUrls.RequestDiagnostics} />} />
+      </Routes>
+    );
+  }
+  return <AppRoutesAfterAnalyticsConsentAnswered />;
+}
```

Comment:
What happened to using the route guard?

If we do do it this way: Wouldn't it make more sense to flip it: creating a separate component for the RequestDiagnosticsPage, and leave the existing routes here?

```tsx
if (!hasResponded) return <AppRoutesBeforeAnalyticsConsentAnswered />;

return <AllTheRoutesAsTheyWereBefore>...
```

The exception is the Request Diagnostics routes, so I'd make a component for that, not the rest of the entire app's routing.

Reply:
> Great that you've changed the page tracking events, this is an improvement. Have a search for .page( though, as in some places we still call this manually.

Thank you :) Do you mean in general in the codebase or as part of this PR? Code unrelated to this feature seems out of scope. This PR has a `.page()` call for the analytics consent page, given the “catch all” routes’ `.page()` seen below is not yet loaded when the consent screen loads.

https://github.com/hirosystems/stacks-wallet-web/blob/8414602e1954bf1d7678be866acc9a39149a79cf/src/app/routes/app-routes.tsx#L51

> What happened to using the route guard?

Using the route guard caused page views to still be picked up by the above before users answered the analytics consent. We would have ended up collecting any routes the users manually navigated to before they responded to the analytics consent, meaning we would not be making the data collection “airtight”, i.e. strictly contained to the analytics page & response prior to user acceptance.

> If we do do it this way: Wouldn't it make more sense to flip it: creating a separate component for the RequestDiagnosticsPage, and leave the existing routes here?

This is the approach taken in an earlier commit, albeit rendering the elements directly. It seemed from an earlier [discussion](https://github.com/hirosystems/stacks-wallet-web/pull/2793#discussion_r1016698921) that we should not follow this pattern, and in any case it doesn’t work with the above catch all `.page()` hook.

> More generally, if this pattern has advantages for this case [...]

It not so much it having advantages, as it is fulfilling the objectives at hand,
- From the feature: strictly ensure only views and responses to consent are collected prior to acceptance
- From earlier feedback: keep as close to the existing patterns as possible

I’m sure there are better ways of doing this, although I can’t see how we’d go about it without, for example, refactoring how we collect page views to collect only what is expected. Incidentally, the current implementation will collect not just page views, but any random string the user types in the URL.

> But for example, why don't we [...]

Seems that it’s a matter of convention: what conventions do we have and follow, and to what extent do the conventions we adopt allow us to implement the features we want? In this PR, I tried my best to follow existing conventions while implementing the feature 😇

Reply:
Sorry if this wasn't clear.

```ts
useEffect(() => void analytics.page('view', `${pathname}`), [analytics, pathname]);
```

This effect improves the page tracking, but in some places page tracking is handled manually, which means this addition will double log page views. As this PR introduces this behaviour, I think it should be fixed before merging.
[Example 1](https://github.com/hirosystems/stacks-wallet-web/blob/5033f1b544be1142e20ab1ce7b26582dc2af9db3/src/app/pages/onboarding/set-password/set-password.tsx#L48-L50). [Example 2](https://github.com/hirosystems/stacks-wallet-web/blob/5033f1b544be1142e20ab1ce7b26582dc2af9db3/src/app/pages/view-secret-key/view-secret-key.tsx#L28-L30). All that's required is removing these page tracking events. 

The other point was just a small one on component organisation, rather than anything relating to a previous conversation. 
As a developer, when I come to `<AppRoutes />` I expect to see the app's routes. But now, the main responsibility of the component is handling the edge case regarding the request diagnostics page. It makes more sense to create an abstraction the edge case, and move that to its own component, rather than the entire app's routes. Here's a clearer example:

```tsx
function AppRoutesBeforeUserHasConsented() {
  return (
    <Routes>
      <Route path={RouteUrls.RequestDiagnostics} element={<AllowDiagnosticsPage />} />
      <Route path="*" element={<Navigate replace to={RouteUrls.RequestDiagnostics} />} />
    </Routes>
  );
}
export function AppRoutes() {
  const hasStateRehydrated = useHasStateRehydrated();
  const hasResponded = useHasUserRespondedToAnalyticsConsent();

  if (!hasStateRehydrated) return <LoadingSpinner />;

  if (!hasResponded) {
    return <AppRoutesBeforeUserHasConsented />
  }

  return <AllTheAppsRoutesAsComponentNameSuggests />;
}
```

Reply:
Ok, I'll re-arrange app routes